client/domain/CorrectnessTestObjectFactory.js

```diff
@@ -31,6 +31,22 @@ tie.factory('CorrectnessTestObjectFactory', [
     };
 
     CorrectnessTest.prototype.matchesOutput = function(output) {
+    // If allowOutputs should be arrays and learner's code returns an array
+    // Then the if statement will check if learner's output matches
+    // any of arrays in the allowedOutputs
+      if (this._allowedOutputs.length > 0 && 
+        this._allowedOutputs[0] instanceof Array &&
+        output instanceof Array) { 
+      // Stringify the arrays and then compare
+        var stringifiedOutput = JSON.stringify(output);
+        for (var i = 0, l = this._allowedOutputs.length; i < l; i++) {
```

Comment:
shouldn't this be this._allowedOutputs.some() ? since `this._allowedOutputs` can have multiple allowed outputs and learner should be able to pass if any of outputs matches?